pandas/_libs/window.pyx

```diff
@@ -354,6 +354,42 @@ def get_window_indexer(values, win, minp, index, closed,
     return indexer.get_data()
 
 
+cdef class WindowIterator:
```

Comment:
Not opposed to using cython, but curious why this can't be implemented directly on the `class _Window` base class?

Reply:
It can be implemented directly, though, we'd have to call libwindow's `get_indexer`, I thought creating this iterator inside the cython file might be better for that reason.

Reply:
I think it should be okay to import `get_indexer` directly and use it to slice the `DataFrames`/`Series`

Reply:
Done!

I did some performence tests and the performance loss is not significant. (A loss of around ~5%)